File: skills/astra-toolkit/clients/typescript/examples/collections-insert-one-id-uuid.ts
import { DataAPIClient, UUID } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db(process.env.ASTRA_DB_API_ENDPOINT!, {
  token: process.env.ASTRA_DB_APPLICATION_TOKEN!,
});
const collection = database.collection("**COLLECTION_NAME**");

// Insert a document into the collection
(async function () {
  const result = await collection.insertOne({
    _id: UUID.v7(),
    name: "Jane Doe",
  });
})();

File: skills/astra-toolkit/clients/python/examples/collections-insert-one-id-uuid.py
import os
from astrapy import DataAPIClient
from astrapy.ids import UUID

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    os.environ["ASTRA_DB_API_ENDPOINT"], token=os.environ["ASTRA_DB_APPLICATION_TOKEN"]
)
collection = database.get_collection("**COLLECTION_NAME**")

# Insert a document into the collection
result = collection.insert_one(
    {
        "_id": UUID("1ef2e42c-1fdb-6ad6-aae4-e84679831739"),
        "name": "Jane Doe",
    },
)
